src/components/body/components/chatRoom/components/addMessage/AddMessage.tsx

```diff
@@ -0,0 +1,104 @@
+//libs
+import {
+  useState,
+  useCallback,
+  useRef,
+  MouseEvent,
+  KeyboardEvent,
+  ChangeEvent,
+} from "react";
+
+//hooks
+import { useUserId } from "../../../../../../hooks/UserContext";
+
+//type
+import { Action, User } from "../../../../../../types/types";
+
+//constant
+import { ACTION } from "../../../../../../constants";
+
+//style
+import styles from "./AddMessage.module.css";
+
+//utils
+import { formatTime } from "../../../../../../utils/getDate";
+import { getDay } from "../../../../../../utils/getDate";
+
+type AddMessageProps = {
+  activeTeamMate: User | undefined;
+  onAction: (action: Action) => void;
+};
+
+export const AddMessage = ({ activeTeamMate, onAction }: AddMessageProps) => {
+  const userId = useUserId();
+  const href = useRef<HTMLTextAreaElement>(null);
+  const [message, setMessage] = useState("");
+
+  const handleChange = useCallback((e: ChangeEvent<HTMLTextAreaElement>) => {
+    if (e.target.value !== "\n") setMessage(e.target.value);
+  }, []);
+
+  const autosize = useCallback(() => {
+    if (href.current) {
+      href.current.style.cssText = "height:auto; padding:0";
+      href.current.style.cssText = "height:" + href.current.scrollHeight + "px";
+    }
+  }, []);
+
+  const handleAddMessage = useCallback(
+    (e: MouseEvent | KeyboardEvent) => {
+      autosize();
+
+      if (
+        (("key" in e && e.key === "Enter") || e.type === "click") &&
+        message.trim().length
+      ) {
+        if (href.current) href.current.style.cssText = "height: 17px";
+
+        onAction({
+          type: ACTION.ADD_MESSAGE,
+          newMessage: {
+            id: crypto.randomUUID(),
```

Comment:
use uuid v4.
https://www.npmjs.com/package/uuid

Also, this id should be generated at the back-end. In response to mutation, call API should return the new message info with id.